lib/ui/view/edit_profile/edit_profile_viewmodel.dart

```diff
@@ -36,164 +34,56 @@ class EditProfileViewModel extends BaseViewModel {
+    String? memId = storageService.getString(StorageKeys.currentMemberID);
     //TODO CHange these links to there rightful values once they can be updated
-    //I interchanged most calls because at the point of writing they did not have their patch call
-    String link =
-        'organizations/61459d8e62688da5302acdb1/members/614729a2f41cb684cc531ac7/photo';
 
-    final editData = {
-      'image_url': _name,
-    };
-    final res = await _api.patch(link, data: editData, headers: {
-      'Authorization':
-          'Bearer ${storageService.getString(StorageKeys.currentSessionToken)}'
-    });
-    String link3 =
-        'organizations/61459d8e62688da5302acdb1/members/614729a2f41cb684cc531ac7/status';
+    String profileEndPoint = 'organizations/$orgId/members/$memId/profile';
 
-    final editData3 = {'status': _status};
-     await _api.patch(link3, data: editData3, headers: {
+    final profileData = {
+      'bio': _status,
+      'first_name': _name,
+      'display_name': _display,
+      'phone': _phone
+    };
+    final editResponse =
+        await _api.patch(profileEndPoint, data: profileData, headers: {
       'Authorization':
           'Bearer ${storageService.getString(StorageKeys.currentSessionToken)}'
     });
-
     final snackbar = locator<SnackbarService>();
-    if (res!.statusCode == 200) {
+
+    if (editResponse!.statusCode == 200) {
       snackbar.showCustomSnackBar(
-        duration: const Duration(seconds: 5),
-        variant: SnackbarType.success,
-        message: ''' Profile Update Was Successful'''
-            ''' Please Close and Open The Edit Profile Page twice to see changes ''',
-      );
-      exitPage();
+          duration: const Duration(seconds: 5),
+          variant: SnackbarType.success,
+          message: ''' Profile Update Was Successful''');
+      await GetUserProfile().currentUser();
+      _navigationService.back();
     } else {
       snackbar.showCustomSnackBar(
         duration: const Duration(seconds: 3),
         variant: SnackbarType.failure,
         message: ''' Profile Update failed''',
       );
     }
+  }
 
-    String link1 =
-        'organizations/61459d8e62688da5302acdb1/members/614729a2f41cb684cc531ac7/profile';
-
-    final editData1 = {
-      'data': _display,
-    };
-    await _api.patch(link1, data: editData1, headers: {
-      'Authorization':
-          'Bearer ${storageService.getString(StorageKeys.currentSessionToken)}'
-    });
-
-    String link2 =
-        'organizations/61459d8e62688da5302acdb1/members/614729a2f41cb684cc531ac7/phone';
-
-    final editData2 = {
-      'phone': _phone,
-    };
-     await _api.patch(link2, data: editData2, headers: {
-      'Authorization':
-          'Bearer ${storageService.getString(StorageKeys.currentSessionToken)}'
-    });
+  @override
+  Future futureToRun() async {
+    setBusy(true);
+    userData = await GetUserProfile().currentUser();
+    setBusy(false);
+    throw UnimplementedError();
```

Comment:
Why is this throwing and error?

Reply:
Unused import

Reply:
Yes, but now the the function is implemented, so you can remove the ```throw UnimplementerError();``` 
If it gives an error the you have to return a value